### PR TITLE
Add args skip failed, ignore missing

### PR DESCRIPTION
**Pull request by T35R6braPwgDJKq**
add an option for skipping failed builds which is otherwise only via config file accessible
and one for ignoring missing aur packages

also contains a testcase for the latter

### CODE REVIEW

**Comment by actionless** [comment]
sorry for the late reply - i was offline

thank you for your contribution! i'll review it tonight

**Comment by actionless** [comment]
i see CI is failing: https://github.com/actionless/pikaur/runs/5337615344?check_suite_focus=true

File: pikaur/help_cli.py [hunk 1]
```diff
@@ -88,6 +89,9 @@ def cli_print_help() -> None:
             ('', '--ignore-outofdate', translate(
                 "ignore AUR packages' updates which marked 'outofdate'"
             )),
+            ('', '--ignore-mssing-aur', translate(
```
Comment (actionless) [inline pikaur/help_cli.py:92]: typo here

File: pikaur_test/test_errors.py [hunk 2]
```diff
@@ -29,6 +29,32 @@ def test_install_not_found(self):
             not_existing_pkg_name
         )
 
+    def test_install_not_found_ignore(self):
```
Comment (actionless) [inline pikaur_test/test_errors.py:32]: i can't understand from these two testcases what's the real life usecase scenario for this flag - could you please elaborate more on it?

Comment (T35R6braPwgDJKq) [reply pikaur_test/test_errors.py:32]: If you have a list of packages that you want to install, it does not break when packages do not exist (anymore)

e.g. `pikaur -S linux notexisting` breaks, `pikaur -S linux notexisting --ignore-missing-aur` installs `linux` and ignores the not existing `notexisting`

The other one is just a flag that you can set in config

File: pikaur/help_cli.py (hunk 1 above)
Comment (T35R6braPwgDJKq) [reply pikaur/help_cli.py]: fixed

**Comment by T35R6braPwgDJKq** [comment]
fixed the CI issue

File: pikaur_test/test_errors.py (hunk 2 above)
Comment (actionless) [reply pikaur_test/test_errors.py:32]: and why it need to be speicifc for aur, not just "ignore-missing-packages" ?

Comment (actionless) [reply pikaur_test/test_errors.py:32]: i don't see any practical usecase for ignoring only aur or only repo packages missing - so i suggest one common flag to just ignore any missing packages

Comment (T35R6braPwgDJKq) [reply pikaur_test/test_errors.py:32]: ok, i changed it to the general case and added the check before the other exit(6)

**Comment by T35R6braPwgDJKq** [comment]
Yes, but:
```
pikaur/install_info_fetcher.py:376:4: R0912: Too many branches (13/12) (too-many-branches)
```
It fails due to the added check of the argument which introduced another `if`. To resolve that correctly the whole function would have to be split in order to split the amounts of branches to different functions.

Id rather suggest to just allow that once.

File: pikaur/install_info_fetcher.py [hunk 3]
```diff
@@ -349,7 +349,7 @@ def get_repo_pkgs_info(self) -> None:
                 self.repo_packages_install_info.append(pkg_update)
             else:
                 self.thirdparty_repo_packages_install_info.append(pkg_update)
-
+
```
Comment (actionless) [inline pikaur/install_info_fetcher.py:352]: this

Comment (T35R6braPwgDJKq) [reply pikaur/install_info_fetcher.py]: yes, I always have to wait for the CI to process it :D

Comment (actionless) [reply pikaur/install_info_fetcher.py]: you can run `./maintenance_scripts/lint.sh` locally

File: pikaur/args.py [hunk 4]
```diff
@@ -53,11 +53,13 @@ def get_pikaur_bool_opts() -> ArgSchema:
         (None, 'nodiff', PikaurConfig().review.NoDiff.get_bool()),
         (None, 'rebuild', None),
         (None, 'dynamic-users', PikaurConfig().build.AlwaysUseDynamicUsers.get_bool()),
+        (None, 'skip-failed-build', PikaurConfig().build.SkipFailedBuild.get_bool()),
```
Comment (actionless) [inline pikaur/args.py:56]: please also add test for this option as well, you could create/use 'broken' PKGBUILD inside `pikaur_test` directory for that

**Pull request closed without merging**